valueside: fix missing support for CITEXT

yuzefovich opened the pull request:
We forgot to add support for CITEXT in MarshalLegacy value encoding. We forgot to add support for CITEXT in MarshalLegacy value encoding. (This encoding type is used when multiple column families are present, and the CITEXT is included not in the zeroth column family.)

No stable release with this bug, so we omit the release note.

Fixes: #150389.
Release none: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/150391)
<!-- Reviewable:end -->

DrewKimball reviewed (approved):
<img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>

> Reviewed 2 of 2 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/150391#-:-OVMBLgeAZj1tUv9NHUB:bnfp4nl)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @paulniziolek and @rytaft)

___
*[`pkg/sql/logictest/testdata/logic_test/citext` line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/150391#-OVMB4SRDXzQJWyu75FY:-OVMB4SRDXzQJWyu75FZ:b-g4rxnl) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7183140717538d83438983af919fecd96562e2d/pkg/sql/logictest/testdata/logic_test/citext#L26)):*
> ```
>   c CITEXT,
>   FAMILY (k),
>   FAMILY (c)
> ```

Feels like we should still have a test with the CITEXT column in the first column family, though I guess that might be covered by the composite test case at the end of the file.


<!-- Sent from Reviewable.io -->

yuzefovich committed:
valueside: fix missing support for CITEXT

We forgot to add support for CITEXT in MarshalLegacy value encoding.
(This encoding type is used when multiple column families are present,
and the CITEXT is included not in the zeroth column family.)

No stable release with this bug, so we omit the release note.

Release none: None

yuzefovich reviewed (commented):
TFTRs!

bors r+

> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/150391#-:-OVO6CsVEDF36hgEjNYK:b-tlx7ko)* status: <span>:shipit:</span> complete! 0 of 0 LGTMs obtained (and 1 stale) (waiting on @DrewKimball and @rytaft)

___
*[`pkg/sql/logictest/testdata/logic_test/citext` line 26 at r1](https://reviewable.io/reviews/cockroachdb/cockroach/150391#-OVMB4SRDXzQJWyu75FY:-OVO67Dk0imp2FFDdrOk:b3dklb9) ([raw file](https://github.com/cockroachdb/cockroach/blob/c7183140717538d83438983af919fecd96562e2d/pkg/sql/logictest/testdata/logic_test/citext#L26)):*
<details><summary><i>Previously, DrewKimball (Drew Kimball) wrote…</i></summary><blockquote>

Feels like we should still have a test with the CITEXT column in the first column family, though I guess that might be covered by the composite test case at the end of the file.
</blockquote></details>

Fair point, added a separate subtest to keep single family case.


<!-- Sent from Reviewable.io -->
